app/controllers/validation/custom_label_destroy_parameters.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Validation
+  class CustomLabelDestroyParameters
+    include ActiveModel::Validations
+
+    attr_accessor :label_id, :custom_label_list
+
+    validates :label_id, presence: true
+    validate :label_id_range?
+    validate :label_deleted?
+
+    def label_id_range?
+      # there can be only 5 custom labels
+      errors.add(:label_id, ':Invalid custom label id') unless
+        label_id.to_i.between?(1, 5)
+    end
+
+    def label_deleted?
+      # check if the label has already been deleted
+      custom_label_list.each do |label|
+        next unless label.id == label_id
+        if label.display_label == ''
+          errors.add(:label_id, ':Label with this id does not exist')
```

Comment:
so if i understood you earlier when we spoke, it sounds like we are in fact showing all custom labels to api consumers on GET to this endpoint.  i agree we want to catch this case and respond to it, but it feels weird saying it was "deleted" here, if a label with that id _does_ technically "exist" in their view were they to run a GET against this endpoint.  is that true or did i miss something?  if that is the case how would you feel about changing this to "label is already

Reply:
mm, yea, I know ....so per RM API, a custom label is considered non-existing if the "displayLabel" attribute is "", sure, our consumers would not know that as it is not intuitive.. I can change the error message to say that "label with this id has been deleted already".